cabra/src/pages/components/SelectBox.tsx

```diff
@@ -0,0 +1,39 @@
+import Select, { SingleValue } from "react-select";
+
+import { useState } from "react";
+
+export interface PaginationOption {
+  readonly value: number;
+  readonly label: string;
+}
+
+const paginationOptions: PaginationOption[] = [
+  { value: 1, label: "1" },
+  { value: 2, label: "2" },
+  { value: 5, label: "5" },
+  { value: 10, label: "10" },
+  { value: 20, label: "20" },
+  { value: 100, label: "100" },
+];
+
+export default function SelectBox(props: {
+  parentCallback: (arg0: number) => void;
+}) {
+  const [selectedOption, setSelectedOption] =
+    useState<PaginationOption | null>();
+
+  const handle = (e: SingleValue<PaginationOption | null>) => {
+    setSelectedOption(e);
+    if (e != null) {
+      props.parentCallback(e.value);
+    }
+  };
+
+  return (
+    <Select<PaginationOption>
+      defaultValue={selectedOption}
+      onChange={handle}
+      options={paginationOptions}
+    />
+  );
+}
```

Comment:
I'm not sure if this component is needed. It has its own state that isn't really used for anything and is just there to run callbacks for `TasksListPaginator`. I think you can move the options to `TasksListPaginator` and directly use the `Select` with the `itemsPerPage` state.

```jsx
// Somewhere in TasksListPaginator.tsx

const DEFAULT_ITEMS_PER_PAGE = 5;
// ...
<Select
  onChange={(selected) =>
    setItemsPerPage(selected?.value ?? DEFAULT_ITEMS_PER_PAGE)
  }
  options={paginationOptions}
  value={paginationOptions.find(({ value }) => value === itemsPerPage)}
/>
```